tests/test_ajd.py

```diff
@@ -1,73 +1,88 @@
+from conftest import get_covmats, get_covmats_params
 from numpy.testing import assert_array_equal
 import numpy as np
-from numpy.testing import assert_allclose
 import pytest
+from pytest import approx
 
 from pyriemann.utils.ajd import rjd, ajd_pham, uwedge, _get_normalized_weight
 
 
-def generate_cov(Nt, Ne):
-    """Generate a set of cavariances matrices for test purpose"""
-    rs = np.random.RandomState(1234)
-    diags = 2.0 + 0.1 * rs.randn(Nt, Ne)
-    A = 2*rs.rand(Ne, Ne) - 1
-    A /= np.atleast_2d(np.sqrt(np.sum(A**2, 1))).T
-    covmats = np.empty((Nt, Ne, Ne))
-    for i in range(Nt):
-        covmats[i] = np.dot(np.dot(A, np.diag(diags[i])), A.T)
-    return covmats, diags, A
-
-
-def test_get_normalized_weight():
+def test_get_normalized_weight(get_covmats):
     """Test get_normalized_weight"""
-    Nt = 100
-    covmats, diags, A = generate_cov(Nt, 3)
+    n_trials, n_channels = 5, 3
+    covmats = get_covmats(n_trials, n_channels)
     w = _get_normalized_weight(None, covmats)
-    assert np.isclose(np.sum(w), 1., atol=1e-10)
+    assert np.sum(w) == approx(1.0, abs=1e-10)
+
 
+def test_get_normalized_weight_length(get_covmats):
+    n_trials, n_channels = 5, 3
+    covmats = get_covmats(n_trials, n_channels)
+    w = _get_normalized_weight(None, covmats)
     with pytest.raises(ValueError):  # not same length
-        _get_normalized_weight(w[:Nt//2], covmats)
+        _get_normalized_weight(w[: n_trials // 2], covmats)
+
+
+def test_get_normalized_weight_pos(get_covmats):
+    n_trials, n_channels = 5, 3
+    covmats = get_covmats(n_trials, n_channels)
+    w = _get_normalized_weight(None, covmats)
     with pytest.raises(ValueError):  # not strictly positive weight
         w[0] = 0
         _get_normalized_weight(w, covmats)
 
 
-def test_rjd():
-    """Test rjd"""
-    covmats, _, _ = generate_cov(100, 3)
+@pytest.mark.parametrize("ajd", [rjd, ajd_pham])
+def test_ajd_shape(ajd, get_covmats):
+    n_trials, n_channels = 5, 3
+    covmats = get_covmats(n_trials, n_channels)
     V, D = rjd(covmats)
-    assert V.shape == (3, 3)
-    assert D.shape == (100, 3, 3)
+    assert V.shape == (n_channels, n_channels)
+    assert D.shape == (n_trials, n_channels, n_channels)
 
 
-def test_pham():
+def test_pham(get_covmats):
     """Test pham's ajd"""
-    Nt = 100
-    covmats, diags, A = generate_cov(Nt, 3)
+    n_trials, n_channels = 5, 3
+    covmats = get_covmats(n_trials, n_channels)
     V, D = ajd_pham(covmats)
 
-    w = 5 * np.ones(Nt)
+    w = n_trials * np.ones(n_trials)
```

Comment:
It should be `w = any_random_positive_number * np.ones(n_trials)`.
Choosing `any_random_positive_number = n_trials` seems a misleading choice.

Reply:
This remark is valide for the tests below.

Reply:
got it